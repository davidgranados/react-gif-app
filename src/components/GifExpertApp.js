import React, { useState } from "react";
import AddCategory from "./AddCategory";
import GifGrid from "./GifGrid";
import PropTypes from "prop-types";

const GifExpertApp = ({defaultCategories}) => {
  const [categories, setCategories] = useState(defaultCategories);
  const onAddCategory = (newCategory) => {
    setCategories((prevState) => [newCategory, ...prevState]);
  };
  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory onAddCategory={onAddCategory} />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

GifExpertApp.propTypes = {
  defaultCategories: PropTypes.array,
}
GifExpertApp.defaultProps  = {
  defaultCategories: [],
}

export default GifExpertApp;
