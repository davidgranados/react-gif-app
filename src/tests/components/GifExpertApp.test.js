import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../../components/GifExpertApp";

describe("Tests on GifExpertApp", () => {
  test("should rended correctly", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should render a categories list", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid")).toHaveLength(categories.length);
  });
});
