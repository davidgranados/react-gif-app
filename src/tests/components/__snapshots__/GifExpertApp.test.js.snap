// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifExpertApp should rended correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    onAddCategory={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Tests on GifExpertApp should render a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    onAddCategory={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
